File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/domain/Invoice.java
package root.proiect_mpp.domain;

public class Invoice implements Entity<Integer> {
    private int id;
    private int clientId;

    // Class Constructors //

    public Invoice() {
        this.id = 0;
        this.clientId = 0;
    }

    public Invoice(int clientId) {
        this.id = 0;
        this.clientId = clientId;
    }

    public Invoice(int id, int clientId) {
        this.id = id;
        this.clientId = clientId;
    }

    // Getters & Setters //

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public int getClientId() {
        return clientId;
    }

    public void setClientId(int clientId) {
        this.clientId = clientId;
    }


    // toString & other functions //

    @Override
    public String toString() {
        return "Invoice{" +
                "id=" + id +
                ", clientId=" + clientId +
                '}';
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/model/Invoice.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model
{
    [Serializable]
    public class Invoice : Entity<int>
    {
        private int clientId;

        // Class Constructors //

        public Invoice() : base(0)
        {
            this.clientId = 0;
        }

        public Invoice(int clientId) : base(0)
        {
            this.clientId = clientId;
        }

        public Invoice(int id, int clientId) : base(id)
        {
            this.clientId = clientId;
        }

        // Getters & Setters //

        public int ClientId
        {
            get { return this.clientId; }
            set { this.clientId = value; }
        }

        // toString & other functions //

        public override string ToString()
        {
            return "Invoice{" +
                "id=" + base.ID +
                ", clientId=" + clientId +
                '}';
        }
    }
}
